fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java

```diff
@@ -679,13 +710,16 @@ private Collection<Long> partitionPrune(PartitionInfo partitionInfo,
         if (partitionInfo.getType() == PartitionType.RANGE) {
             if (isPointQuery() && partitionInfo.getPartitionColumns().size() == 1) {
                 // short circuit, a quick path to find partition
-                ColumnRange filterRange = columnNameToRange.get(partitionInfo.getPartitionColumns().get(0).getName());
-                LiteralExpr lowerBound = filterRange.getRangeSet().get().asRanges().stream()
-                        .findFirst().get().lowerEndpoint().getValue();
-                LiteralExpr upperBound = filterRange.getRangeSet().get().asRanges().stream()
-                        .findFirst().get().upperEndpoint().getValue();
+                Column col = partitionInfo.getPartitionColumns().get(0);
+                // ColumnRange filterRange = columnNameToRange.get(col.getName());
+                Set<Range<ColumnBound>> filterRanges =
+                        columnNameToRange.get(col.getName()).getRangeSet().get().asRanges();
                 cachedPartitionPruner.update(keyItemMap);
-                return cachedPartitionPruner.prune(lowerBound, upperBound);
+                Collection<Long> prunedPartIDs = cachedPartitionPruner.prune(filterRanges);
+                partitionCol2PartitionID.putIfAbsent(
```

Comment:
cachedPartitionPruner.prune(filterRanges, partitionCol2PartitionID) add partitionCol2PartitionID to prune params, and call  putIfAbsent in prune function